dragonfly/room2d.py

```diff
@@ -1063,40 +1075,36 @@ def _honeybee_plenums(self, hb_room, tolerance=0.01):
                     If there isn't enough space between the Story floor_height and
                     the Room2D floor_height, this item will be None.
         """
-
-        hb_rooms = [None, None]
-
-        try:
-            parent = self.parent
-        except AttributeError:
-            print('Cannot add plenums to the {} Room because the parent Story has '
-                  'not been set. This is required to derive the plenum '
-                  'height.'.format(self.identifier))
-            raise
-
-        ceil_plenum_height = \
-            parent.floor_to_floor_height - self.floor_to_ceiling_height
+        # check to be sure that the room2d has a parent story
+        hb_rooms = []
+        if not self.has_parent:
+            raise AttributeError(
+                'Cannot add plenums to the "{}" Room because the parent Story has '
+                'not been set. This is required to derive the plenum '
+                'height.'.format(self.identifier))
+
+        parent = self.parent
+        parent_ceiling = parent.floor_height + parent.floor_to_floor_height
+        ceil_plenum_height = parent_ceiling - self.ceiling_height
         floor_plenum_height = self.floor_height - parent.floor_height
 
         if ceil_plenum_height > tolerance:
             ceil_plenum = self._honeybee_plenum(
                 ceil_plenum_height, plenum_type="ceiling")
-            # Overwrite the previous hb_room ceiling BC
-            try:
-                hb_room[-1].boundary_condition = bcs.adiabatic
-            except AttributeError:
-                pass
-            hb_rooms[0] = ceil_plenum
+            # Set the plenum and the rooms to be adjacent to one another
+            hb_room[-1].set_adjacency(ceil_plenum[0], tolerance)
+            hb_rooms.append(ceil_plenum)
```

Comment:
@chriswmackey I think we have to specify the index that we are inserting the plenums here, not just append it. The way the return arguments are set up, the `ceil_plenum` should the second item in the list of HB Rooms returned by the to_honeybee method, and the `floor_plenum` should be the third item. 

The reasoning is that the index number indicates if it's a floor or ceiling plenum.  Here's the docstrings:
```
         hb_room -- A honeybee-core Room representing the dragonfly Room2D. If the
                add_plenum argument is True this item will be a list of honeybee-core
                Rooms, with the hb_room as the first item, and two additional items:
                     * ceil_plenum -- A honeybee-core Room representing the ceiling
                        plenum. If there isn't enough space between the Story
                        floor_to_floor_height and the Room2D floor_to_ceiling height,
                        this item will be None.
                    * floor_plenum -- A honeybee-core Room representing the floor plenum.
                        If there isn't enough space between the Story floor_height and
                        the Room2D floor_height, this item will be None.
```

So if we just append to a list, and the ceil_plenum doesn't exist, the floor_plenum will be at index 1. Alternatively we can just return a list of plenums, and use the identifier or some other means to confirm if it's a ceiling or floor plenum.

Reply:
@saeranv ,
Yes, this was one of the tweaks I made. I don't want to have a list with None's in it as this will mess up the methods that use the Room.2Dto_honeybee() higher in the dragonfly schema hierarchy. And this is much more important than being able to identify floor vs. ceiling plenums by index.